File: StarMines/Data/Python/red.py
def behave():
    b = Behavior()
     
    if b.getRegisterInt(1) == 0:
        init(b)
    
    if b.getRegisterInt(0) == 0:
        switch(b)
    move(b)

    #check calls

    call = b.getCall()
    while call:
        if call.contents.label == b'collision':
            b.sendToCaller(call.contents.caller, b'infect', None, 0)
        call = b.getCall()

def move(b):
    pos = b.position()
    x = pos.contents.x
    y = pos.contents.y

    if x > 770:
        x = 770
        b.setPosition(x,y)
        switch(b)
    if x < 0:
        x = 0
        b.setPosition(x,y)
        switch(b)
    if y > 570:
        y = 570
        b.setPosition(x,y)
        switch(b)
    if y < 0:
        y = 0
        b.setPosition(x,y)
        switch(b)
    
    counter = b.getRegisterInt(0)
    counter -= 1
    b.setRegisterInt(0, counter)


def switch(b):
    b.setRegisterInt(0, random.randint(20,50))
    x = 0.0
    y = 0.0
    r = random.randint(0,3)
    if r == 0:
        x = -1.0
    if r == 1:
        x = 1.0
    if r == 2:
        y = -1.0
    if r == 3:
        y = 1.0

    b.setDirection(x,y)
    

def init(b):
    b.setRegisterInt(1,1)
    b.setSpeed(6)

behave()

File: StarMines/TestData/Pandemic/Lua/Red.lua
function Red_change(bc)
	local r = math.random(1,4)
	if r == 1 then bc:setDir(1,0) end
	if r == 2 then bc:setDir(-1,0) end
	if r == 3 then bc:setDir(0,1) end
	if r == 4 then bc:setDir(0,-1) end
	bc:setRegInt(0, math.random(20,50))
end

function Red_bounds(bc)
	local pos = bc:position()
	local x = pos.x
	local y = pos.y
	if x > 770 then x = 770 Red_change(bc) end
	if x < 0 then x = 0 Red_change(bc) end
	if y > 570 then y = 570 Red_change(bc) end
	if y < 0 then y = 0 Red_change(bc) end
	bc:setPosition(x,y)
end

function Red_move(bc)
	local counter = bc:getRegInt(0)
	if counter == 0 then
		Red_change(bc)
	else
		bc:setRegInt(0, counter - 1)
	end
	
	Red_bounds(bc)
end

function Red(bc)
	bc:setSpeed(5)
	Red_move(bc)
end